Delcam.ProductInterface/Automation.cs

```diff
@@ -215,22 +215,24 @@ protected void CloseAllInstancesOf(string name)
         {
             Console.WriteLine("Closing all instances...");
 
-            // Find all the processes with the passed in name and kill them
-            foreach (Process process in Process.GetProcessesByName(name))
+            var processesRunning = true;
+            var startTime = DateTime.UtcNow;
+            while (processesRunning && DateTime.UtcNow - startTime < TimeSpan.FromSeconds(10))
             {
-                try
-                {
-                    process.Kill();
-                }
-                catch
+                foreach (Process process in Process.GetProcessesByName(name))
                 {
+                    try
+                    {
+                        process.Kill();
+                    }
+                    catch
+                    {
+                    }
                 }
-                process.WaitForExit(10000);
-                if (process.HasExited == false)
-                {
-                    throw new Exception("Failed to close all instances of " + name);
-                }
+
```

Comment:
Don't you want raise an exception if processesRunning > 0 after the ellapsed time?

Reply:
Sensible suggestion.  Modification made.